### PR TITLE
Experiment: Allocate ID strings all at once.

### PR DESCRIPTION
**Pull request by markshannon**
This is an experiment to allocate all the `_Py_IDENTIFIER` strings in one struct.
Ignoring the generated code, there is a net reduction in code size.
Performance [is about the same](https://gist.github.com/markshannon/439562ea6040d91bb0d78157959daf63), but this should help work on startup time.

### CODE REVIEW

**Comment by tiran** [comment]
@markshannon What do you think about using macros? https://github.com/tiran/cpython/pull/new/pr29432-macro reduce the file size of core_objects.c from 240k to 50k.

File: Modules/_datetimemodule.c [hunk 1]
```diff
@@ -5131,7 +5119,9 @@ datetime_fromtimestamp(PyObject *cls, PyObject *args, PyObject *kw)
                                    tzinfo);
     if (self != NULL && tzinfo != Py_None) {
         /* Convert UTC to tzinfo's zone. */
-        self = _PyObject_CallMethodId(tzinfo, &PyId_fromutc, "N", self);
+        PyObject *tmp = self;
+        self = PyObject_CallMethodOneArg(tzinfo, _Py_ID(fromutc), tmp);
+        Py_DECREF(tmp);
```
Comment (ericsnowcurrently) [inline Modules/_datetimemodule.c:5124]: This is a bug fix too?

File: Modules/_bisectmodule.c [hunk 2]
```diff
@@ -5,6 +5,7 @@ Converted to C by Dmitry Vasiliev (dima at hlabs.spb.ru).
 
 #define PY_SSIZE_T_CLEAN
 #include "Python.h"
+#include "core_objects.h"
```
Comment (ericsnowcurrently) [inline Modules/_bisectmodule.c:8]: Is this needed?

File: Objects/typeobject.c [hunk 3]
```diff
@@ -7658,31 +7597,29 @@ slot_tp_setattro(PyObject *self, PyObject *name, PyObject *value)
...
-    _Py_static_string_init("__gt__"),
-    _Py_static_string_init("__ge__"),
+static PyObject *name_op[] = {
+    _Py_ID(__lt__),
+    _Py_ID(__le__),
+    _Py_ID(__eq__),
+    _Py_ID(__ne__),
+    _Py_ID(__gt__),
+    _Py_ID(__ge__),
 };
```
Comment (ericsnowcurrently) [inline Objects/typeobject.c:7623]: This seems like a step backward.

File: Lib/test/test_exceptions.py [hunk 4]
```diff
@@ -1506,7 +1506,6 @@ class C(): pass
         for i in range(1, 20):
             rc, out, err = script_helper.assert_python_failure("-c", code % i)
             self.assertIn(rc, (1, 120))
-            self.assertIn(b'MemoryError', err)
```
Comment (ericsnowcurrently) [inline Lib/test/test_exceptions.py:1509]: Was this intentional?

File: Modules/_bisectmodule.c [hunk 5]
```diff
@@ -131,7 +130,7 @@ _bisect_insort_right_impl(PyObject *module, PyObject *a, PyObject *x,
             return NULL;
     }
     else {
-        result = _PyObject_CallMethodId(a, &PyId_insert, "nO", index, x);
+        result = PyObject_CallMethod(a, "insert", "nO", index, x);
```
Comment (sweeneyde) [inline Modules/_bisectmodule.c:133]: Is there a reason this is using `PyObject_CallMethod` (without the underscore)? There seem to be 10 call sites where `_PyObject_CallMethodId` is converted to `PyObject_CallMethod`.

**Review by ericsnowcurrently — COMMENTED** [review]
Overall, this looks pretty good.  Here are a few comments after a quick skim through the change.

**Comment by markshannon** [comment]
Abandoning this for now.

**Pull request closed without merging**